pages/select_profile_page.py

```diff
@@ -0,0 +1,777 @@
+)
+from utils.data_handling_utils import (
+    check_if_value_exists,
+    get_avg_population_value,
+    get_value_and_delta,
+)
+from config.highlights import highlights
+
+# Setting up themes and fonts
+alt.themes.register("nestafont", nestafont)
+alt.themes.enable("nestafont")
+
+# Loading the data
+distribution_households = dg.get_distribution_households()
+profile_annual_avgs = dg.get_annual_energy_consumption_avg_per_profile()
+elec_hh_data = dg.get_hh_consumption_per_profile(energy_type="electricity")
+gas_hh_data = dg.get_hh_consumption_per_profile(energy_type="gas")
+elec_hh_data_summer = dg.get_hh_consumption_per_profile(
+    energy_type="electricity", season="summer"
+)
+elec_hh_data_winter = dg.get_hh_consumption_per_profile(
+    energy_type="electricity", season="winter"
+)
+gas_hh_data_summer = dg.get_hh_consumption_per_profile(
+    energy_type="gas", season="summer"
+)
+gas_hh_data_winter = dg.get_hh_consumption_per_profile(
+    energy_type="gas", season="winter"
+)
+contextual_data = dg.get_contextual_information()
+
+
+def initial_setup() -> int:
+    """
+    This function sets up the initial configuration for the 'Select a profile' page.
+
+    Returns:
+        int: The selected profile number.
+    """
+    # List of profiles to select from in the format "Profile X: Y% of the population"
+    options = [
+        f"Profile {i}: {(int(round(distribution_households["perc_of_households"].iloc[i-1])))}% of the population"
+        for i in range(1, 11)
+    ]
+
+    # Create a selectbox for profile selection centered in the page (making use of columns)
+    col_1, profile_selector_col, col_2 = st.columns((1, 2, 1))
+    with profile_selector_col:
+        profile_selector = st.selectbox(
+            label="Choose one energy-use profile", options=options, width=500
+        )
+
+    # Extract the profile number from the selected option
+    profile_selector = int(
+        profile_selector.split()[1].split(":")[0]
+    )  # Extract the profile number from the string
+
+    # Provide a paragraph with some highlights about this profile
+    st.markdown(
+        f"##### Overview: profile {profile_selector} " + highlights[profile_selector]
+    )
+    return profile_selector
+
+
+def setup_profile_main_metrics(profile_selector: int):
+    """
+    Sets up the section for displaying information about the selected energy-use profile
+    including:
+    - Number and proportion of households in the profile
+    - Average annual electricity and gas consumption
+
+    Args:
+        profile_selector (int): profile number selected by the user.
+    """
+
+    # METRICS AT THE TOP
+    profile_nums_filtered = distribution_households[
+        distribution_households["profile"] == profile_selector
+    ].iloc[0]
+
+    col_1, n_households, perc_households, col_2 = st.columns(4)
+
+    n_households.metric(
+        label="Number of households",
+        value=str(int(profile_nums_filtered["number_of_households"])),
```

Comment:
```suggestion
        value=f"{int(profile_nums_filtered['number_of_households']):,}",
```

For improved readability, I suggest adding a comma separator for thousands. If you agree, it would also be useful to do this for the average annual electricity and gas consumption metric boxes.

Reply:
this is a really great suggestion! Thank you! And I didn't know about such an easy way to format thousands. 😍